### ISSUE TITLE
Enable documentation updates without doing full releases

### ISSUE THREAD

**Issue by knolleary** [flowforge/flowforge#1200]
### Description

Currently the documentation on the website is built from the latest release of the `flowforge` repository. This makes it hard to make documentation updates without doing a new release.

We should be able to make multiple doc updates every day with as little friction as possible.

It isn't appropriate to pull the content from the `main` branch as that will include docs for unreleased work in the current milestone.

We don't want to pull the docs into its own repository as that will break the link between features and the docs getting merged together.

## Updated Proposal

 - All doc updates are made in the `main` branch as normal via PR
 - If a doc update is also intended to be deployed live, the PR should be labelled `docs-publish`
 - Change our maintenance process so we have a single branch to publish maintenance releases from - rather than create a new maintenance branch with each release. Call this branch `maintenance` (`maint` is a more succint option, but that's one character off `main` and could cause accidents)
 - Update the website automation to pull the docs from the `maintenance` branch
 - Update release automation of `flowforge/flowforge` to force-push `main` to `maintenance` when a new release is created from the `main` branch
 - A new automation is created that is triggered whenever a PR is merged with the `docs-publish` label.
      - It creates a new PR that cherry-picks the commits to the `maintenance` branch and assigns the PR for review by the author of the original PR. (Decided it would be safer to have a PR step, rather than direct cherry-picking to avoid merge-conflict headaches)

<details><summary>Old Proposal</summary>
 - Create a `live-docs` branch. This will be the branch the website always pulls from.
 - All doc updates are made in the `main` branch as normal via PR
 - If a doc update is also intended to be deployed live, the PR should be labelled `live-docs`
 - A new automation is created that is triggered whenever a PR is merged with that label.
      - It cherry-picks the commits to the `live-docs` branch and triggers a website rebuild.
      - (maybe) It cherry-picks the commits to the current maintenance release branch
 - The release automation script, when operating on the `main` branch (not maintenance release branch), force pushes `main` to `live-docs` and triggers website rebuild

If someone forgets to add the `live-docs` label to the docs PR, they will have to manually PR the update to the `live-docs` branch themselves and trigger a website rebuild.
</details>

**Comment by knolleary** [comment]
I have updated the proposal above following discussions with Ben and ZJ.

TL;DR: change our maintenance process to use a single branch for maintenance releases, and have the website always pull from that branch. That will mean just merging a docs change to the branch will get it published to the website, without requiring a full maintenance release.

**Comment by knolleary** [comment]
Once these PRs are merged, I think we're done on this task. The hard part is we cannot test the final integration until we do an actual release. Will keep this issue open until we're done on 1.1 and have seen the website pull from maintenance as expected

- [ ] https://github.com/flowforge/flowforge/pull/1300
- [ ] https://github.com/flowforge/website/pull/258

**Comment by knolleary** [comment]
I believe all the pieces are in place now for this. The 1.1.0 release was published and the website rebuilt from the `maintenance` branch successfully.
